helpers/Physics.h

```diff
@@ -0,0 +1,380 @@
+#ifndef _HELPERS_PHYSICS_H_
+#define _HELPERS_PHYSICS_H_
+
+#include "../include/z64.h"
+#include "Matrix.h"
+
+typedef struct {
+	s32   J; // Amount of strand limbs + 1
+	Vec3f headPos;
+	Vec3s headRot;
+	f32*  lengthList;
+	struct {
+		Vec3f scale;  // Gfx scale
+		u32   dlist;
+		u8    segID;  // For matrix
+		u8    noDraw; /* If there needs to be calculations
+		                 to get in position before drawing */
+	} gfx;
+	struct {
+		s32    numVec;      // num of how many vec are used from vecList
+		Vec3f* vecList;     // "collision" spheres
+		f32    dist;        // XYZ distance from limb to vecList member
+	} spheres;
+	struct {
+		Vec2f rotLimit;     // DEG, limits rot to next limb in main calc
+		f32   drawYawLimit; // DEG, limits rot on draw, smoothens output
+	} constraint;
+	
+	/* Physics */
+	f32 gravity;
+	f32 floor;        // world.pos.y, won't go through this
+	f32 maxVel;       // Clamps velocity value
+	f32 velStep;      // Values below 1.0f will give it spring like motion
+	f32 velMult;      // Control the power of velocity
+	
+	/* Shape Push */
+	s32   numPush;    // How many limbs will be smoothed with push
+	Vec3f push;       // direction Z, pushes based on rot[0]
+	f32   pushMult;   // How much the pushing fill affect
+	Vec2f headRotAdd; // DEG, relative, used to control push direction
+} PhysicsStrand;
+
+void Physics_SetHead(PhysicsStrand* params) {
+    MtxF mtx;
+    Vec3f zero = { 0 };
+	
+    Matrix_Get(&mtx);
+    func_800D20CC(&mtx, &params->headRot, 0);
```

Comment:
```suggestion
    Vec3f zero = { 0 };
	
    func_800D20CC(&sCurrentMatrix, &params->headRot, 0);
```
there's no need to copy the matrix, `func_800D20CC` doesn't edit the matrix argument